File: QuirkyCarRepairApi/QuirkyCarRepair.BLL/Areas/CarService/DTO/GetServiceOrderPage.cs
using QuirkyCarRepair.DAL.Areas.Shared.Enums;

namespace QuirkyCarRepair.BLL.Areas.CarService.DTO
{
    public class GetServiceOrderPage
    {
        public int Page { get; set; }
        public int PageSize { get; set; }

        public bool AnyDate { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }

        public string? SortBy { get; set; }
        public SortDirection SortDirection { get; set; }

        public List<OrderStatus>? OrderStates { get; set; }
    }
}

File: quirky-car-repair/src/api/service/getServiceOrderPage.js
import {BACK_END_URL} from "../../constans/backEndUrl.js";
import axios from "axios";

export async function getServiceOrderPage(data) {
    const user = sessionStorage["user"] ? JSON.parse(sessionStorage["user"]) : [];
    const body={
        page: data && data.page ? data.page : 1,
        pageSize: data && data.pageSize ? data.pageSize : 10,
        anyDate: data && data.anyDate !==undefined ? data.anyDate : true,
        fromDate: data && data.fromDate ? data.fromDate : null,
        toDate: data && data.toDate ? data.toDate : null,
        sortBy: "string",
        sortDirection:0,
        orderStates:data && data.orderStates ? data.orderStates : []
    };

    try {
        const response = await axios.post(`${BACK_END_URL}/CarService/GetServiceOrderPage`, body, {
            headers: {
                Authorization: `Bearer ${user.token}`,
            },
        });
        return { success: true, data: response.data};
    } catch (error) {
        console.error('Błąd podczas pobierania danych:', error);
        return { success: false, message: 'Błąd podczas pobierania danych' };
    }
}